openapi.yml

```diff
@@ -1253,12 +1791,421 @@ components:
+          items:
+            $ref: "#/components/schemas/FileContractRevision"
+        storageProofs:
+          type: array
+          description: List of storage proofs asserting the storage of data for file contracts.
+          items:
+            $ref: "#/components/schemas/StorageProof"
+        siafundInputs:
+          type: array
+          description: List of Siafund inputs spent in the transaction.
+          items:
+            $ref: "#/components/schemas/SiafundInput"
+        siafundOutputs:
+          type: array
+          description: List of Siafund outputs created by the transaction.
+          items:
+            $ref: "#/components/schemas/SiafundOutput"
+        minerFees:
+          type: array
+          description: List of miner fees included in the transaction.
+          items:
+            $ref: "#/components/schemas/Currency"
+        arbitraryData:
+          type: array
+          description: Arbitrary binary data included in the transaction.
+          items:
+            type: string
+            format: byte
+        signatures:
+          type: array
+          description: List of cryptographic signatures verifying the transaction.
+          items:
+            $ref: "#/components/schemas/TransactionSignature"
+
+    TransactionID:
+      allOf:
+        - $ref: "#/components/schemas/Hash256"
+        - description: Unique identifier for a transaction.
+
+    TransactionSignature:
+      type: object
+      properties:
+        parentID:
+          allOf:
+            - $ref: "#/components/schemas/Hash256"
+            - description: The ID of the transaction being signed
+        publicKeyIndex:
+          type: integer
+          format: uint64
+          description: The index of the public key used to sign the transaction
+        timelock:
+          type: integer
+          format: uint64
+          description: The block height at which the output can be spent
+        coveredFields:
+          allOf:
+            - $ref: "#/components/schemas/CoveredFields"
+            - description: Indicates which fields of the transaction are covered by the signature
+        signature:
+          type: string
+          format: byte
+          description: The signature of the transaction
+
+    UnlockConditions:
+      type: object
+      properties:
+        timelock:
+          type: integer
+          format: uint64
+          description: The block height at which the output can be spent
+        publicKeys:
+          type: array
+          items:
+            $ref: "#/components/schemas/UnlockKey"
+        signaturesRequired:
+          type: integer
+          format: uint64
+          description: The number of signatures required to spend the output
+
+    UnlockKey:
+      type: object
+      properties:
+        algorithm:
+          type: string
+          format: bytes
+        key:
+          type: string
+          format: bytes
```

Comment:
I'd add patterns for both algorithm and key. The algorithm is always the same anyway and the key is hex encoded I think? (better to double check that)

Right now the example just says algorithm: "string" and key: "string" when looking at the swagger editor which isn't nice.